packages/components/src/tree/tree.tsx

```diff
@@ -69,109 +90,189 @@ export function Tree<T, EL extends HTMLElement = HTMLElement>(props: TreeProps<T
         itemSize,
         ...scrollListProps
     } = props;
-    const [openItems, updateOpenItems] = useStateControls(openItemsControls, []);
-    const [focused, updateFocused] = useStateControls(focusControl, undefined);
-    const { items, depths } = useMemo(
-        () => getItems({ data, getChildren, getId, openItems }),
-        [data, getChildren, getId, openItems]
+    const [openItemIds, setOpenItemIds] = useStateControls(openItemsControls, []);
+    const [focusedItemId, focus] = useStateControls(focusControl, undefined);
+    const [, select] = useStateControls(scrollListProps.selectionControl, undefined);
```

Comment:
not really sure what it's used for, you update the selection but doesn't use it? is it needed?

Reply:
Yeah, it's the inner library's control over controls passed to it. User owns focused and selected states, but stylable components can interject with built-in logic